File: 2021/tests/exercises/test_exercise_14.py
from adventofcode.exercises import exercise_14
from adventofcode import utils

TEST_DATA = """NNCB

CH -> B
HH -> N
CB -> H
NH -> C
HB -> C
HC -> B
HN -> C
NN -> C
BH -> H
NC -> B
NB -> B
BN -> B
BB -> N
BC -> B
CC -> N
CN -> C"""


def test_example_on_exercise_a():
    assert exercise_14.run_a(TEST_DATA) == 1588


def test_returns_correct_answer_on_exercise_a():
    input_data = utils.load_data(exercise_14.EXERCISE)
    assert exercise_14.run_a(input_data) == 2712


def test_example_on_exercise_b():
    assert exercise_14.run_b(TEST_DATA) == 2188189693529


def test_returns_correct_answer_on_exercise_b():
    input_data = utils.load_data(exercise_14.EXERCISE)
    assert exercise_14.run_b(input_data) == 8336623059567

File: 2024/adventofcode/tests/test_exercise_14.rs
use adventofcode::exercises::exercise_14;
use adventofcode::exercises::utils;

const TEST_DATA: &str = "p=0,4 v=3,-3
p=6,3 v=-1,-3
p=10,3 v=-1,2
p=2,0 v=2,-1
p=0,0 v=1,3
p=3,0 v=-2,-2
p=7,6 v=-1,-3
p=3,0 v=-1,-2
p=9,3 v=2,3
p=7,3 v=-1,2
p=2,4 v=2,-3
p=9,5 v=-3,-3";

#[test]
fn test_a_returns_correct_answer_for_test_data() {
    assert_eq!(exercise_14::run_a(TEST_DATA, 11, 7), 12);
}

#[test]
fn test_a_returns_correct_answer_on_exercise_a() {
    let input_data = utils::load_data(exercise_14::EXERCISE);
    assert_eq!(exercise_14::run_a(&input_data, 101, 103), 222208000);
}

#[test]
fn test_b_returns_correct_answer_on_exercise_b() {
    let input_data = utils::load_data(exercise_14::EXERCISE);
    assert_eq!(exercise_14::run_b(&input_data), 7623);
}
